services/src/main/scala/cromwell/services/MetadataServiceActor.scala

```diff
@@ -34,3 +37,22 @@ object MetadataServiceActor {
   }
   */
 }
+
+object MetadataServiceActorImplicits {
+  implicit class EnhancedServiceRegistryActorForMetadata(val actor: ActorRef) extends AnyVal {
+    def pushWdlValueMetadata(metadataKey: MetadataKey, output: WdlValue): Unit = output match {
+      case WdlArray(_, valueSeq) =>
+        val zippedSeq = valueSeq.zipWithIndex
+        zippedSeq foreach { case (value, index) => actor.pushWdlValueMetadata(metadataKey.copy(key = s"${metadataKey.key}[$index]"), value) }
+      case WdlMap(_, valueMap) =>
+        valueMap foreach { case (key, value) => actor.pushWdlValueMetadata(metadataKey.copy(key = metadataKey.key + s":$key"), value) }
+      case value =>
+        actor ! PutMetadataAction(MetadataEvent(metadataKey, MetadataValue(value)))
```

Comment:
That is not going to do what you think it will, because value is a `WdlValue`, and the metadata "type evaluator" doesn't know how to interpret wdl values. So we either need to 
`case ...` all the `WdlType`s here and use the underlying value or make the MetadataType able to "detect" `WdlValue`s real types.